join-statements/joinstatements.md

```diff
@@ -0,0 +1,423 @@
+
+> SELECT t.column1 
+> FROM table_name AS t
+> WHERE t.id = 20
+> INNER JOIN table2_name;
+
+or to represent an aliased column:
+
+> SELECT AVG(column1) AS average_score, name
+> FROM table_name
+> WHERE alias1 = 20
+> GROUP BY name
+> ORDER BY average_score DESC;
+
+This would allow for more readable queries, and perhaps economize keystrokes for typing if the query is particularly long and the aliased column or table is used multiple times in the query.
+
+5) 
+SELECT p.name, c.salary, c.vacation_days 
+FROM professor AS p JOIN
+compensation AS c ON p.id = c.professor_id;
+
+6)
+`NATURAL JOIN` would be ideal to use in cases where column names have the exact same name in both tables. For example, if we have a `fantasy_gamer` table with the the columns `real_name`, `email_address`, and another table called `charachters` with the columns `charachter_name` and `email_address`, we would be able to create the following query:
+
+```
+SELECT fantasy_gamer.real_name
+FROM fantasy_gamer
+NATURAL JOIN charachters;
+
+```
+
+this query would yield the real names of users with emails that match in both `fantasy_gamer` table and the `charachters` table.
+
+This is because there exists an `email_address` column in both the `fantasy_gamer` table, as well as the `charachters` table.
+
+7) 
+
+I created these tables:
+```
+contract_business=# SELECT *
+contract_business-# FROM shifts
+contract_business-# ORDER BY shift_id ;
+
+ shift_id | day_of_week | shift | employee_id 
+----------+-------------+-------+-------------
+        0 | Sunday      |     0 |           3
+        1 | Sunday      |     1 |           4
+        2 | Sunday      |     2 |           5
+        3 | Monday      |     0 |           2
+        4 | Monday      |     1 |           0
+        5 | Monday      |     2 |           1
+        6 | Tuesday     |     0 |           2
+        7 | Tuesday     |     1 |           0
+        8 | Tuesday     |     2 |           1
+        9 | Wednesday   |     0 |           2
+       10 | Wednesday   |     1 |           0
+       11 | Wednesday   |     2 |           1
+       12 | Thursday    |     0 |           2
+       13 | Thursday    |     1 |           0
+       14 | Thursday    |     2 |           1
+       15 | Friday      |     0 |           2
+       16 | Friday      |     1 |           0
+       17 | Friday      |     2 |           1
+       18 | Saturday    |     0 |           3
+       19 | Saturday    |     1 |           4
+       20 | Saturday    |     2 |           5
+(21 rows)
+```
+
+and...
+
+```
+contract_business=# SELECT * FROM employees;
+
+ employee_id |  name   | gender 
+-------------+---------+--------
+           0 | Xavier  | M
+           1 | Maria   | F
+           2 | Roger   | M
+           3 | Justine | F
+           4 | Omar    | M
+           5 | Tammy   | F
+```
+
+So for the first part, this is listing all employees and all shifts:
+
+```
+SELECT a.name, b.day_of_week, b.shift
+FROM employees AS a
```

Comment:
e and s would be more appropriate aliasign

Reply:
done